error.go

```diff
@@ -105,3 +110,27 @@ func IsStatusTooManyRequests(err error) bool {
 func IsRequestFailed(err error) bool {
 	return IsClientErrorType(err, RequestFailed)
 }
+
+type ApiError struct {
+	HTTPStatus int `json:"httpStatus"`
+	Messages   []struct {
+		ErrorCode string `json:"errorCode"`
+		Message   string `json:"message"`
+	} `json:"messages"`
+}
+
+func (e ApiError) Error() string {
+	return e.String()
+}
+
+func (e ApiError) String() string {
+	toReturn := fmt.Sprintf("HTTP Status: %s \n%s", fmt.Sprint(e.HTTPStatus), "Error Messages:")
```

Comment:
Why don't you use `%d` here? Apart from that, remove the space before the newline.

Reply:
this was just moved from req.go without further review. Fixed